src/Internet/Logging/Loom_MongoDB/Loom_MongoDB.cpp

```diff
@@ -225,35 +225,33 @@ void Loom_MongoDB::loadConfigFromJSON(char* json){
     port = 0;
     
     /* We should check if any parameter is null */
-    if(!doc["broker"].isNull())
+    if(!doc["broker"].isNull() && 
+    !doc["database"].isNull() && 
+    !doc["username"].isNull() && 
+    !doc["password"].isNull() && 
+    !doc["port"].isNull() && 
+    strlen(projectServer) > 0){
```

Comment:
I believe this still does not allow the project field to be optional, it checks if everything is not null and if the projectServer variable has a length greater than 0 at that time which will actually never be true because that value likely hasn't even been set yet